homeassistant/components/vacuum/xiaomi_miio.py

```diff
@@ -383,3 +401,37 @@ def update(self):
             _LOGGER.error("Got OSError while fetching the state: %s", exc)
         except DeviceException as exc:
             _LOGGER.warning("Got exception while fetching the state: %s", exc)
+
+    async def async_clean_zone_start(self,
+                                     zone,
+                                     reps: int = 1,
+                                     reps_template: str = None,
+                                     test: bool = 0):
+        """Cleaning selected area for the number of reps indicated."""
+        if reps_template is not None:
+            _LOGGER.debug("Using template reps")
+            reps_template.hass = self.hass
+            reps = reps_template.async_render()
+            try:
+                reps = int(reps)
+            except ValueError:
+                reps = 1
+                _LOGGER.debug("Failed to cast template reps. Set default > 1")
+        _LOGGER.debug("Start clean zone: %s Reps: %s", zone, reps)
+        from miio import DeviceException
+        try:
+            for _zone in zone:
+                _LOGGER.debug("original zone: %s", _zone)
+                _zone.append(reps)
+                _LOGGER.debug("zone with reps: %s", _zone)
+        except OSError as exc:
+            _LOGGER.error("Got OSError while append reps to zone: %s", exc)
+        except DeviceException as exc:
+            _LOGGER.warning("Got exception while append reps to zone: %s", exc)
+        """If test, bypass exec command on vacuum."""
+        if not test:
+            await self._try_command(
+                "Unable to send command to the vacuum: %s",
+                self._vacuum.raw_command, ZONE_CLEAN_COMMAND, zone)
```

Comment:
Instead of using raw_command, you should prefer (and extend if necessary) the python-miio provided command: https://github.com/rytilahti/python-miio/blob/master/miio/vacuum.py#L93

Reply:
I'll study it for next release